pkg/manifests/nmstateconfig.go

```diff
@@ -99,3 +116,26 @@ func ProcessNMStateConfig(infraEnv aiv1beta1.InfraEnv) ([]*models.HostStaticNetw
 	})
 	return staticNetworkConfig, nil
 }
+
+// Retrieve the first IP from the user provided NMStateConfig yaml file to set as node0 IP
+func GetNodeZeroIP() string {
+	config, err := getNMStateConfig()
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+	var nmconfig NMStateConfig
+	err = yaml.Unmarshal(config.Spec.NetConfig.Raw, &nmconfig)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+	var nodeZeroIP string
+	if nmconfig.Interfaces[0].IPV4.Address != nil {
+		nodeZeroIP = nmconfig.Interfaces[0].IPV4.Address[0].IP
+	} else {
+		nodeZeroIP = nmconfig.Interfaces[0].IPV6.Address[0].IP
```

Comment:
Minor, but would be useful to have a sanity check for *nmconfig.Interfaces[0].IPV6.Address* not set and exit with error message. Also could check if Interfaces is set and that resulting nodeZeroIP is a valid IP using net.ParseIP(nodeZeroIP)